sycl/doc/extensions/experimental/sycl_ext_oneapi_matrix/sycl_ext_oneapi_matrix.asciidoc

```diff
@@ -20,247 +19,399 @@
+
+== Backend support status
+This extension is currently implemented in {dpcpp} only for devices
+that contain a matrix hardware, specifically Intel(R) Advanced Matrix
+Extensions (Intel(R) AMX), Intel(R) Xe Matrix Extensions (Intel(R)
+XMX) and Nvidia(R) Tensor Cores.
+
+The `joint_matrix` type and the `joint_matrix_mad` function are
+optional kernel features as defined in section 5.7 of the core SYCL
+specification.  Each device supports only certain values for the `M`,
+`N`, and `K` template parameters and only certain types for the `Ta`,
+`Tb`, and `Tc` template parameters. Applications can use the query API
+in `matrix_params` or `get_info<experimental::info::device::matrix>`
+to determine the set of legal parameters for each device.  If the
+application submits a kernel using an unsupported `joint_matrix` type
+or calls `joint_matrix_mad` with an unsupported combination, the
+implementation throws a synchronous exception with the
+`errc::kernel_not_supported` error code as described in section 5.7.
+
+== Overview
+Joint matrix is a SYCL extension for matrix hardware programming. It
+unifies targets like Intel AMX in CPUs, Intel XMX in Intel GPUs and
+Nvidia Tensor Cores. This provides a portable and performant API for
+users who want to build their own neural networks applications,
+perform custom optimzations, or experiment with new operations in a
+timely and performing manner.
+
+== Specification
+
+=== Feature test macro
 
-## New `joint_matrix` class
-We introduce a new class called `joint_matrix`. The user needs to specify the group memory scope, the type of the elements, the shape, the matrix use, and the memory layout of the matrix. This results in the following description:
+This extension provides a feature-test macro as described in the core SYCL
+specification. An implementation supporting this extension must predefine
+the macro `SYCL_EXT_ONEAPI_MATRIX` to one of the values defined in the
+table below. Applications can test for the existence of this macro to
+determine if the implementation supports this feature, or applications
+can test the macro's value to determine which of the extension's
+features the implementation supports.
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|The APIs of this experimental extension are not versioned, so the
+ feature-test macro always has this value.
+|===
+
+=== New `joint_matrix` class
+We introduce a new class called `joint_matrix`. The user needs to
+specify the group memory scope, the type of the elements, the shape,
+the matrix use, and the memory layout of the matrix. This results in
+the following description:
 
 ```c++
 namespace sycl::ext::oneapi::experimental::matrix {
+
 template <typename Group, typename T, use Use, size_t Rows, size_t Cols,
-          layout Layout = layout::dynamic>
-struct joint_matrix {
-    joint_matrix() {}
-};
-}
-```
+          layout Layout = (Use == use::accumulator) ?
+          layout::dynamic : /*unspecified*/ >
+struct joint_matrix;
```

Comment:
I just noticed that you removed the body of the `joint_matrix` structure from the specification.  I think we need to specify what constructors and assignment operators it supports.  This would be a minimal set, which is probably all we need:

```
template <typename Group, typename T, use Use, size_t Rows, size_t Cols,
          layout Layout = (Use == use::accumulator) ?
          layout::dynamic : /*unspecified*/ >
struct joint_matrix {
  joint_matrix();
  joint_matrix(const joint_matrix &);
  joint_matrix &operator=(const joint_matrix &);
};
```

The spec should also say whether these constructors / copy-assignment-operator are "group functions" that must be encountered in convergent code.  This is somewhat related to `joint_matrix_apply`.  If that function is not a group function, then the constructors / copy-assignment-operator probably should not be either.

Reply:
We don't support the assignment operator and copy constructor.
You cannot do something like:
joint_matrix<> A;
joint_matrix<> B;
A=B;
If we would enable that, we will have to provide a separate API for this.
In the meantime, use joint_matrix_apply to assign/copy elements of the two matrices.